src/java/org/jivesoftware/openfire/XMPPServer.java

```diff
@@ -141,6 +145,13 @@
     private boolean shuttingDown;
     private XMPPServerInfoImpl xmppServerInfo;
 
+    /*
+     * Provider for fine-grained access control, based on:
+     * "clearances" (user authorization data) and
+     * "labels" (per-item, user-supplied, access control data).
+     */
+    private AccessControlDecisionFunction accessControlDecisionFunction;
+
```

Comment:
Could this go in a Module, instead of straight into XmppServer? It'd be good to separate concerns, where possible.